clang/lib/CodeGen/CGCall.cpp

```diff
@@ -1531,7 +1531,8 @@ void ClangToLLVMArgMapping::construct(const ASTContext &Context,
     case ABIArgInfo::Direct: {
       // FIXME: handle sseregparm someday...
       llvm::StructType *STy = dyn_cast<llvm::StructType>(AI.getCoerceToType());
-      if (AI.isDirect() && AI.getCanBeFlattened() && STy) {
+      if (AI.isDirect() && AI.getCanBeFlattened() && STy &&
+          !STy->containsHomogeneousScalableVectorTypes()) {
```

Comment:
Try to handle this in `RISCVABIInfo::classifyArgumentType` by `setCanBeFlattened(false)` rather than change here,